### PR TITLE
feat: added apm

### PR DESCRIPTION
**Pull request by cl3m0**
# Description

Added Elastic Application Performance Monitoring (to be configured)

### CODE REVIEW

File: src/licensing/main.py [hunk 1]
```diff
@@ -78,6 +78,22 @@ async def log_requests(request, call_next):
...
 
 
+# add application performance monitoring middleware
+apm = make_apm_client(
+    {
+        "SERVICE_NAME": f"licensing-{settings.segment}",
+        "SECRET_TOKEN": settings.apm_secret_token,
+        "SERVER_URL": settings.apm_url,
+        "ENVIRONMENT": settings.segment,
+        "TRANSACTIONS_IGNORE_PATTERNS": ["^OPTIONS", "/status"],
```
Comment (saurabh2590) [inline src/licensing/main.py:88]: Please check status route.

File: docker-compose.yml [hunk 2]
```diff
@@ -16,7 +16,7 @@ services:
       - DATABASE_NAME=licm
     depends_on:
       - postgres
-  
+
   postgres:
     image: postgres:15.1-alpine
```
Comment (saurabh2590) [inline docker-compose.yml:21]: Currently, in the cloud we don't have Postgres 15, I recommend using version 14 to avoid surprises in the remote environment. (Not part of this PR)

File: k8s/lib/config.ts [hunk 3]
```diff
@@ -26,36 +26,52 @@ import {
...
   [Segment.DEV00]: {
+    SEGMENT: Segment.DEV00,
     LOG_HANDLER: "console",
     LOG_LEVEL: LogLevel.DEBUG,
     DATABASE_USER: "postgres",
     DATABASE_HOST: "localhost",
     DATABASE_PORT: "5432",
     DATABASE_NAME: "licm",
+    APM_URL: "https://apm.bettermarks.com",
+    APM_ENABLED: true,
```
Comment (saurabh2590) [inline k8s/lib/config.ts:49]: Question:  By default, we have kept `false` for the dev00 segments? You can keep it on for dev00 but we can consider turning off for dev00.

Comment (cl3m0) [reply k8s/lib/config.ts:49]: I wanted to test it on dev00 and later switch it off... if ok

**Review by saurabh2590 — APPROVED** [review]
Looks ok to me; however, we need to check why APM had issues and if we have a performance problem. :)
